File: core/src/main/java/pcb/uwu/amount/derived/electromagnetism/MagneticFlux.java
package pcb.uwu.amount.derived.electromagnetism;

import pcb.uwu.core.BigDecimalAmount;
import pcb.uwu.core.CompositeUnitAmount;
import pcb.uwu.core.Magnitude;
import pcb.uwu.core.UnitAmount;
import pcb.uwu.unit.derived.electromagnetism.MagneticFluxUnit;

import java.math.BigDecimal;
import java.math.MathContext;

import static pcb.uwu.utils.UnitAmountUtils.getAmountIn;

public class MagneticFlux extends CompositeUnitAmount<MagneticFluxUnit> {

	// region constructors

	public MagneticFlux(Number value, MagneticFluxUnit unit) {
		super(value, unit);
	}

	public MagneticFlux(Number value, Magnitude magnitude, MagneticFluxUnit unit) {
		super(value, magnitude, unit);
	}

	public MagneticFlux(String value, MagneticFluxUnit unit) {
		super(value, unit);
	}

	public MagneticFlux(String value, Magnitude magnitude, MagneticFluxUnit unit) {
		super(value, magnitude, unit);
	}

	public MagneticFlux(BigDecimal value, MagneticFluxUnit unit) {
		super(value, unit);
	}

	public MagneticFlux(BigDecimal value, Magnitude magnitude, MagneticFluxUnit unit) {
		super(value, magnitude, unit);
	}

	public MagneticFlux(BigDecimalAmount amount, MagneticFluxUnit unit) {
		super(amount, unit);
	}

	public MagneticFlux(BigDecimalAmount amount, Magnitude magnitude, MagneticFluxUnit unit) {
		super(amount, magnitude, unit);
	}

	// endregion

	// region implement UnitAmount

	@Override
	public MagneticFlux plus(UnitAmount<MagneticFluxUnit> other, MathContext mathContext) {
		return new MagneticFlux(plusAmount(other, mathContext), getUnit());
	}

	@Override
	public MagneticFlux minus(UnitAmount<MagneticFluxUnit> other, MathContext mathContext) {
		return new MagneticFlux(minusAmount(other, mathContext), getUnit());
	}

	@Override
	public MagneticFlux multipliedBy(BigDecimal other, MathContext mathContext) {
		return new MagneticFlux(multipliedByScalar(other, mathContext), getUnit());
	}

	@Override
	public MagneticFlux dividedBy(BigDecimal other, MathContext mathContext) {
		return new MagneticFlux(dividedByScalar(other, mathContext), getUnit());
	}

	@Override
	public MagneticFlux in(MagneticFluxUnit unit) {
		return new MagneticFlux(getAmountIn(this, unit), unit);
	}

	// endregion
}

File: core/src/main/java/pcb/uwu/amount/derived/electromagnetism/MagneticFlux.kt
package pcb.uwu.amount.derived.electromagnetism

import pcb.uwu.core.CompositeUnitAmount
import pcb.uwu.core.Magnitude
import pcb.uwu.core.Magnitude.NATURAL
import pcb.uwu.core.UnitAmount
import pcb.uwu.unit.derived.electromagnetism.MagneticFluxUnit
import pcb.uwu.utils.UnitAmountUtils
import java.math.BigDecimal
import java.math.MathContext

open class MagneticFlux : CompositeUnitAmount<MagneticFluxUnit>
{
    @JvmOverloads
    constructor(value: Number,
                magnitude: Magnitude = NATURAL,
                unit: MagneticFluxUnit)
            : super(value, magnitude, unit)

    @JvmOverloads
    constructor(value: String,
                magnitude: Magnitude = NATURAL,
                unit: MagneticFluxUnit)
            : super(value, magnitude, unit)

    // region UnitAmount

    override operator fun plus(other: UnitAmount<MagneticFluxUnit>) =
        MagneticFlux(value = this.amount + other.into(this.unit).amount,
                     unit = this.unit)

    override operator fun minus(other: UnitAmount<MagneticFluxUnit>) =
        MagneticFlux(value = this.amount - other.into(this.unit).amount,
                     unit = this.unit)

    override fun multiply(other: BigDecimal, mathContext: MathContext) =
        MagneticFlux(value = UnitAmountUtils.multipliedByScalar(this, other, mathContext),
                     unit = this.unit)

    override fun div(other: BigDecimal, mathContext: MathContext) =
        MagneticFlux(value = UnitAmountUtils.dividedByScalar(this, other, mathContext),
                     unit = this.unit)

    override fun into(unit: MagneticFluxUnit) =
        MagneticFlux(value = UnitAmountUtils.getAmountIn(this, unit),
                     unit = this.unit)

    // endregion
}